common/structs.go

```diff
@@ -100,12 +102,37 @@ type Equipment struct {
 	ReserveAmmo    int
 }
 
+// GrenadeProjectile is a grenade thrown intentionally by a player. It is used to track grenade projectile
+// positions between the time at which they are thrown and until they detonate.
+type GrenadeProjectile struct {
+	EntityID int
+	Weapon   EquipmentElement
+	Thrower  *Player
+	Owner    *Player
+
+	// internalID is used to distinguish different grenades (which potentially have the same, reused entityID) from each other.
+	internalID int64
+
+	Position r3.Vector
+}
+
 // Class returns the class of the equipment.
 // E.g. pistol, smg, heavy etc.
 func (e Equipment) Class() EquipmentClass {
 	return e.Weapon.Class()
 }
 
+// NewGrenadeProjectile creates a grenade projectile and sets.
+func NewGrenadeProjectile() *GrenadeProjectile {
+	// @micvbang TODO: decide if we want to use random ints or a sequence from the parser here.
+	return &GrenadeProjectile{internalID: rand.Int63()}
+}
+
+// InternalID returns the internal id of the grenade, which is used to distinguish grenades that reuse another entity's entity id.
+func (g GrenadeProjectile) InternalID() int64 {
```

Comment:
I think the ID should be renamed to UniqueID or something, otherwise it's implying the library consumer isn't supposed to use it.

Reply:
Good point. I'll try and see if I can find a better name for it. `UniqueID` is not bad, but I'll think about it some more :)

Reply:
I ended up with the suggested `UniqueID` name :slightly_smiling_face: